builder/index.html

```diff
@@ -257,7 +258,7 @@ <h4 class="panel-title">
                         <span class="glyphicon glyphicon-plus-sign" style="margin-right:5px"></span>
                         Add methods
                     </div>
-                    <button ng-disabled="editForm.$pending || editForm.$invalid || CForm.form.$invalid || checkSelectedTechnique()" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
+                    <button ng-disabled="editForm.$pending || editForm.$invalid || CForm.form.$invalid || checkSelectedTechnique() || selectedMethod.advanced_class.indexOf(' ')>0 || CForm.versionForm.$invalid" class="btn btn-default pull-right" ng-click="saveTechnique()">Save</button>
```

Comment:
YOu should not add the check on wetherit contains a space or not, but be based on the fact that Cform.form is $invalid (check advanced class input with a patter)